nmt.go

```diff
@@ -185,10 +185,7 @@ func (n *NamespacedMerkleTree) ProveRange(start, end int) (Proof, error) {
 // ProveNamespace returns a range proof for the given NamespaceID.
 //
 // case 1) If the namespace nID is out of the range of the tree's min and max
-// namespace i.e., (nID < n.minNID) or (n.maxNID < nID) ProveNamespace returns
-// an error and does not generate any range proof, instead it returns an empty
-// Proof with empty nodes and the range (0,0) i.e., Proof.start = 0 and
-// Proof.end = 0 to indicate that this namespace is not contained in the tree.
```

Comment:
I think we just need to remove where we say it returns error, the rest is fine. 

```
case 1) If the namespace nID is out of the range of the tree's min and max
// namespace i.e., (nID < n.minNID) or (n.maxNID < nID) ProveNamespace returns an empty
// Proof with empty nodes and the range (0,0) i.e., Proof.start = 0 and
// Proof.end = 0 to indicate that this namespace is not contained in the tree.
```

Reply:
Sure. IMO it seems excessively verbose to explicitly state the contents of an empty share proof in this comment. I think a more maintainable approach would be to reference "an empty share proof" in this comment. If desired the function that generates an empty share proof can describe why it returns Proof.start = 0 and Proof.end = 0.

Reply:
resolved by https://github.com/celestiaorg/nmt/pull/167/commits/8854743ead3c84817a1cdbf4d94fcb0c5ebc2abb

Reply:
> If desired the function that generates an empty share proof can elaborate on the contents of an empty share proof

Agree, though we need to have it ( "an empty share proof" ) defined somewhere.